docs/deprecation.md

```diff
@@ -0,0 +1,23 @@
+# Deprecation Policy
+
+## User-facing commands and their arguments
+
+Whenever you rename a command or make a previously optional argument required,
+we require that you preserve the old command invocations keeps on working for 6
+months (so 6 releases, since we release monthly) with a deprecation message.
+The message should inform the user that the previous workflow is deprecated
+and to be removed in the future.
+
+## Packaging commands and niche commands
+
+For commands with a niche user audience or something we assume is rarely used
+(we sadly have no data), we take the liberity to remove the old behavior within
+two releases. This means that you can change the old command to immediately
+error.
```

Comment:
Maybe we can mention `jj debug` commands specifically. I think breaking changes in them is always fine. But we can still try to balance it against how hard it is to keep compatibility.

Reply:
> Maybe we can mention `jj debug` commands specifically. I think breaking changes in them is always fine. But we can still try to balance it against how hard it is to keep compatibility.

Since the `jj debug` commands actually are internal APIs, I think they don't even should get mentioned.